api/objectstore.go

```diff
@@ -0,0 +1,193 @@
+package api
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/longhorn/longhorn-manager/datastore"
+	longhorn "github.com/longhorn/longhorn-manager/k8s/pkg/apis/longhorn/v1beta2"
+	"github.com/longhorn/longhorn-manager/util"
+	"github.com/pkg/errors"
+	"github.com/rancher/go-rancher/api"
+	"github.com/rancher/go-rancher/client"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (s *Server) ObjectStoreList(rw http.ResponseWriter, req *http.Request) (err error) {
+	apiContext := api.GetApiContext(req)
+	col, err := s.objectStoreList(apiContext)
+	if err != nil {
+		return err
+	}
+	apiContext.Write(col)
+	return nil
+}
+
+func (s *Server) objectStoreList(apiContext *api.ApiContext) (*client.GenericCollection, error) {
+	list, err := s.m.ListObjectStoresSorted()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to list object stores")
+	}
+
+	data := []interface{}{}
+
+	for _, store := range list {
+		vol, err := s.m.Get(fmt.Sprintf("pv-%v", store.Name))
```

Comment:
Suggesting to move function `genPVName()` and similar function to utils package and call it here

Reply:
To be honest I'm not quite happy with `genPVName` anyways, since it relies a lot on the uniqueness of names. I'd much rather let the controller put some checks in place and/or utilize labels and references to associate the resources with each other.
Although for now this works ok, it seems quite fragile to me, but sure I'll move it to the utils package

Reply:
On the flip side, I am wondering why not use the same ObjectStore name for volume, PV, and PVC? It makes it easier to convert naming between ObectStore <--> volume <--> PV <--> PVC. We are already using the same ObectStore for Deployment/Service/Secret... so I am not sure why we don't do the same for volume/pV/PVC

Reply:
Different K8s objects have different requirements for names:
https://kubernetes.io/docs/concepts/overview/working-with-objects/names/

Therefore a name that works for an `objectstore` does not necessarily work for a PV or PVC. E.g. I found during testing that PVs can generally not be named something that starts with a number like "1-volume", but object stores can.
`genPVName` was the simplest workaround at the time to generate a name that would work for the PV from the name of the object store. It has the drawback that it doesn't do any sanity checks. I think the same applies to the PVC, but it has a different prefix for consistency sake. I'm not sure why the volume's name starts with `pv-` as well, but I remember there being a problem if it doesn't, something like with static provisioning, the name of the volume must match the name of the PV or similar.

What I'd really like to see is that all generated resources have some kind of unique ID in their name like it is the case for deployments:
Deployment name "my-deployment" --> replicaset name "my-deployment-xyzabc" --> pod name "my-deployment-xzyabc-defghi"
This really isn't a critical feature though, as the name of the object store already is unique in the longhorn namespace. It would help a user identify which K8s objects are generated and managed through the controller and which aren't though.

Reply:
ping @m-ildefons

Reply:
Looks like this is implemented now. Revolving